src/cljam/io/vcf/util/normalize.clj

```diff
@@ -0,0 +1,165 @@
+(ns cljam.io.vcf.util.normalize
+  (:require [clojure.string :as cstr]
+            [cljam.io.vcf.util :as vcf-util]
+            [cljam.io.sequence :as io-seq]))
+
+(defn- biallelic-map
+  [m id->number ^long i ^long n-allele]
+  (let [ploidy (or (some-> m :GT vcf-util/parse-genotype count) 2)]
+    (into
+     {}
+     (map
+      (fn [[k v]]
+        (let [n (id->number k)]
+          [k (cond
+               (= k :GT) (vcf-util/biallelic-genotype v (inc i))
+               (= n "A") [(nth v i)]
+               (= n "R") [(first v) (nth v (inc i))]
+               (= n "G") (vcf-util/biallelic-coll ploidy n-allele (inc i) v)
+               :else v)])))
+     m)))
+
+(defn- get-id->number
+  [coll]
+  (into {} (map (juxt (comp keyword :id) :number)) coll))
+
+(defn make-splitter
+  "Returns a function that splits a multiallelic variant into a sequence of
+  biallelic variants."
+  [{:keys [info format]} header]
+  (let [info-id->number (get-id->number info)
+        format-id->number (get-id->number format)
+        samples (map keyword (drop 9 header))]
+    (fn split [{:keys [alt] :as v}]
+      (let [n (count alt)]
+        (if (<= n 1)
+          [v]
+          (->> alt
+               (map-indexed
+                (fn sample-ith-alt [i allele]
+                  (-> v
+                      (assoc :alt [allele])
+                      (update :info biallelic-map info-id->number i n)
+                      (into
+                       (map
+                        (fn [s]
+                          [s (biallelic-map (v s) format-id->number i n)]))
+                       samples))))))))))
+
+(defn same-ref?
+  "Checks if a REF allele matches the reference sequence."
+  [seq-reader {:keys [chr ^long pos ^String ref]}]
+  (let [ref-len (.length ref)
+        ref-region {:chr chr :start pos :end (dec (+ pos ref-len))}
+        ref-seq (io-seq/read-sequence seq-reader ref-region {:mask? false})]
+    (and ref-seq (.equalsIgnoreCase ^String ref-seq ref))))
+
+(defn- regions-before [chr ^long pos ^long window]
```

Comment:
I'm not sure if I read what you meant correctly, but it looks `regions-before` generates `window - 1` bases wide window of regions.

```clojure
(regions-before "chr1" 10 3)
;=> ({:chr "chr1", :start 8, :end 9}
;    {:chr "chr1", :start 6, :end 7}
;    {:chr "chr1", :start 4, :end 5}
;    {:chr "chr1", :start 2, :end 3}
;    {:chr "chr1", :start 1, :end 1})
```

Reply:
Thank you for pointing that out! You're right, they should span exactly `window` bases. Fixed the function and test cases in f21905b.